src/beast/beast/nudb/common.h

```diff
@@ -28,8 +27,31 @@
 namespace beast {
 namespace nudb {
 
+// Commonly used types
+
+enum class file_mode
+{
+    scan,         // read sequential
+    read,         // read random
+    append,       // read random, write append
+    write         // read random, write random
+};
+
+using path_type = std::string;
+
 // All exceptions thrown by nudb are derived
-// from std::exception except for fail_error
+// from std::runtime_error except for fail_error
+
+/** Thrown when a codec fails, e.g. corrupt data. */
+struct codec_error : std::runtime_error
+{
+    template <class String>
+    explicit
+    codec_error (String const& s)
+        : runtime_error(s)
+    {
+    }
+};
 
 /** Base class for all errors thrown by file classes. */
 struct file_error : std::runtime_error
```

Comment:
The constructors below can be replaced with `using std::runtime_error::runtime_error`. I tested in clang, not sure about VS support. (not new code and not asking for a change).